### ISSUE TITLE
`Layout/EmptyLineAfterMagicComment` incorrectly flags multiple magic comments (`typed` / `rbs_inline`) as separate groups

### ISSUE THREAD

**Issue by marcoroth** [rubocop/rubocop#14695]
## Steps to reproduce the problem

Create a Ruby file with the following content:

```ruby
# frozen_string_literal: true
# typed: true
# rbs_inline: enabled

puts "hello"
```

Run RuboCop:

```sh
bundle exec rubocop
```

## Expected behavior

RuboCop should **not register an offense**.

All three comments are magic comments used by common Ruby tooling (Ruby itself, Sorbet, and RBS Inline) and should be treated as a single group at the top of the file, without requiring blank lines between them.

## Actual behavior

RuboCop registers an offense from `Layout/EmptyLineAfterMagicComment`:

```
Offenses:

test_rubocop.rb:3:1: C: [Correctable] Layout/EmptyLineAfterMagicComment: Add an empty line after magic comments.
# rbs_inline: enabled
^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

 and suggests inserting blank lines between each comment, resulting in:

```ruby
# frozen_string_literal: true

# typed: true

# rbs_inline: enabled

puts "hello"
```

## RuboCop version

```
$ bundle exec rubocop -V
1.81.7 (using Parser 3.3.10.0, rubocop-ast 1.48.0, analyzing as Ruby 3.2, running on ruby 3.4.7) [arm64-darwin24]
```

---

If this expected behavior, feel free to close this issue, thank you!
